include/zephyr/bluetooth/classic/hfp_ag.h

```diff
@@ -111,10 +111,10 @@ struct bt_hfp_ag_cb {
 	 *  If this callback is provided it will be called whenever the
 	 *  SCO/eSCO connection gets disconnected.
 	 *
-	 *  @param ag HFP AG object.
-	 *  @param sco_conn SCO/eSCO Connection object.
+	 *  @param conn SCO/eSCO Connection object.
+	 *  @param reason BT_HCI_ERR_* reason for the disconnection.
 	 */
-	void (*sco_disconnected)(struct bt_hfp_ag *ag);
+	void (*sco_disconnected)(struct bt_conn *sco_conn, uint8_t reason);
 
```

Comment:
Is it better to add `struct bt_hfp_ag *ag` too because `sco_connected` callback's parameter has `struct bt_hfp_ag *ag`?

Reply:
OK, Just my thought: (1) The stack can give the `struct bt_hfp_ag *ag` easily, then upper application layer doesn't need to use the `sco_conn` to match the `struct bt_hfp_ag *ag` again in `sco_disconnected` callback. It should be more easy for application to use same callback function to implement multiple instances. (2) It is symmetrical because `sco_connected` has it.